ifrs/core/file_registry.py

```diff
@@ -57,18 +58,32 @@ async def _is_file_registered(self, *, file: models.FileMetadata) -> bool:
         except ResourceNotFoundError:
             return False
 
-        if file == registered_file:
+        # object ID is a UUID generated upon registration, so cannot compare those
+        file_without_object_id = file.dict(exclude={"object_id"})
+        registered_file_without_object_id = registered_file.dict(exclude={"object_id"})
+
+        if file_without_object_id == registered_file_without_object_id:
             return True
 
         raise self.FileUpdateError(file_id=file.file_id)
 
-    async def register_file(self, *, file: models.FileMetadata) -> None:
+    async def register_file(
+        self,
+        *,
+        file: models.FileMetadataBase,
```

Comment:
Would probably be more consistent if this was already `file_without_object_id` to communicate intent and then rename `file_with_object_id` to just `file`

Reply:
Updated in [d5cd92f](https://github.com/ghga-de/internal-file-registry-service/pull/39/commits/d5cd92f7bbfeb9876407a9f94c159b0f5e19d49f)